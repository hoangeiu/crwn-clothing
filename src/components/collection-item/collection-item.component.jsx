import React from "react";
import { connect } from "react-redux";
import { addItem } from "../../redux/cart/cart.actions";
import CustomeButton from "../custom-button/custom-button.component";

import "./collection-item.styles.scss";

const CollectionItem = ({ item, addItem }) => {
  const { id, name, price, imageUrl } = item;
  return (
    <div className="collection-item">
      <div
        className="image"
        style={{
          backgroundImage: `url(${imageUrl})`,
        }}
      />
      <div className="collection-footer">
        <span className="name">{name}</span>
        <span className="price">{price}</span>
      </div>
      <CustomeButton onClick={() => addItem(item)} inverted>
        Add to cart
      </CustomeButton>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item)),
});

export default connect(null, mapDispatchToProps)(CollectionItem);
