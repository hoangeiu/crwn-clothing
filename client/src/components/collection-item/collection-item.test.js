import React from "react";
import { shallow } from "enzyme";

import { CollectionItem } from "./collection-item.component";

describe("CollectionItem component", () => {
  let wrapper;
  let mockAddItem;
  let mockImageUrl = "www.test.com";
  let mockPrice = 300;
  let mockName = "Blue skirt";
  beforeEach(() => {
    mockAddItem = jest.fn();

    const mockProps = {
      item: {
        id: 1,
        name: mockName,
        price: mockPrice,
        imageUrl: mockImageUrl,
      },
      addItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it("should CollectionItem render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call addItem when button is clicked", () => {
    wrapper.find("CustomButton").simulate("click");

    expect(mockAddItem).toHaveBeenCalled();
  });

  it("should render name correctly", () => {
    const name = wrapper.find(".name").text();

    expect(name).toBe(mockName);
  });
});
