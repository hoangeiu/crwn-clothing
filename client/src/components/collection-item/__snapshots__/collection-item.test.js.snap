// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should CollectionItem render without error 1`] = `
<div
  className="collection-item"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(www.test.com)",
      }
    }
  />
  <div
    className="collection-footer"
  >
    <span
      className="name"
    >
      Blue skirt
    </span>
    <span
      className="price"
    >
      300
    </span>
  </div>
  <CustomButton
    inverted={true}
    onClick={[Function]}
  >
    Add to cart
  </CustomButton>
</div>
`;
